app/resources/person_resource.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2022, Schweizer Wanderwege. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+class PersonResource < ApplicationResource
+  primary_endpoint 'people', [:index, :show, :update]
+
+  attribute :first_name, :string
+  attribute :last_name, :string
+  attribute :nickname, :string
+  attribute :company_name, :string
+  attribute :company, :boolean
+  attribute :email, :string
+  attribute :address, :string
+  attribute :zip_code, :string
+  attribute :town, :string
+  attribute :country, :string
+  attribute :gender, :string, readable: :show_details?, writable: :write_details?
+  attribute :birthday, :date, readable: :show_details?, writable: :write_details?
+  attribute :primary_group_id, :integer, except: [:writeable]
+
+  has_many :phone_numbers,
+    link: false,
+    resource: PhoneNumberResource,
+    readable: :show_details_or_public?
+  has_many :social_accounts,
+    link: false,
+    resource: SocialAccountResource,
+    readable: :show_details_or_public?
+  has_many :additional_emails,
+    link: false,
+    resource: AdditionalEmailResource,
+    readable: :show_details_or_public?
+
+  has_many :roles,
+    link: false,
+    resource: RoleResource,
+    readable: :show_full?,
+    writable: false
+
+  filter :updated_at, :datetime, single: true do
+    eq do |scope, value|
+      scope.where(updated_at: value..)
+    end
+  end
+
+  def show_full?(model_instance)
+    can?(:show_full, model_instance)
+  end
+
+  def show_details?(model_instance)
+    can?(:show_details, model_instance)
+  end
+
+  def write_details?
+    # no model_instance method argument is given when writable is called,
+    # so we have to access current entry by controller context
+    can?(:show_details, context.entry)
+  end
+
+  def show_details_or_public(model_instance)
+    show_details?(model_instance) || model_instance.public?
```

Comment:
Ist `model_instance` jetzt eine Person oder eine PhoneNumber? Entweder der erste oder der zweite Teil dieser Bedingung kann nie funktionieren.

Reply:
totally ... hab's in die contactable resources verschoben.